src/rred_reports/reports/auth.py

```diff
@@ -51,4 +97,25 @@ def get_account(self) -> Account:
         Returns:
             Account: exchangelib Account object for an authenticated user
         """
-        return Account(primary_smtp_address=f"{self.settings.send_emails_as}", config=self.get_config(), access_type=DELEGATE)
+        conf = self.get_config()
+        return Account(primary_smtp_address=self.settings.send_emails_as, config=conf, access_type=DELEGATE)
+
+
+def _check_or_set_up_cache():
+    """Set up MSAL token cache and load existing token"""
+    cache = msal.SerializableTokenCache()
+    cache_path = Path("my_cache.bin")
+
+    if cache_path.exists():
+        with cache_path.open("rb") as cache_file:
+            cache.deserialize(cache_file.read())
+
+    def save_cache():
+        if cache.has_state_changed:
+            with cache_path.open("wb") as cache_file:
+                cache_file.write(cache.serialize())
+
+    save_cache()
```

Comment:
Oh interesting. I'm surprised this actually works saving it before we're authenticated. Would have thought you'd have to save the data to the cache file after we've done the manual 2FA flow?